src/language/dictionary/JMDict/Spelling.java

```diff
@@ -14,13 +13,15 @@
     private boolean isKanji;
     private String spelling;
     private Spelling[] dependencies;
+    private List<Spelling> readings;
```

Comment:
I tried comparing the memory use of building vs not building the readings list for each spelling, and also of having a member for the list of readings at all, but couldn't even tell the difference because Java's memory use is so nondeterministic.